pkg/types/docker_conf.go

```diff
@@ -21,10 +23,26 @@ func GetDockerOption(insecureTlsSkip bool, Platform string) (types.DockerOption,
 	if err := env.Parse(&cfg); err != nil {
 		return types.DockerOption{}, xerrors.Errorf("unable to parse environment variables: %w", err)
 	}
+	credentials := make([]types.Credential, 0)
+	users := strings.Split(cfg.UserName, ",")
+	passwords := strings.Split(cfg.Password, ",")
+	for index, user := range users {
+
+		if len(user) > 0 && index < len(passwords) && len(passwords[index]) > 0 {
+			credentials = append(credentials, types.Credential{
+				UserName: strings.TrimSpace(user),
+				Password: strings.TrimSpace(passwords[index]),
+			})
+		}
+	}
```

Comment:
What if comparing the lengths before the loop?

```suggestion
	if len(users) != len(passwords) {
		return types.DockerOption{}, xerrors.New("the length of usernames and passwords must match")
	}
	for index, user := range users {
		credentials = append(credentials, types.Credential{
			UserName: strings.TrimSpace(user),
			Password: strings.TrimSpace(passwords[index]),
		})
	}
```